File: endpoints/src/main/java/com/cromoteca/samples/endpoints/ReviverEndpoint.java
package com.cromoteca.samples.endpoints;

import com.cromoteca.samples.annotations.Nullable;
import dev.hilla.Endpoint;
import java.math.BigInteger;
import java.util.Date;
import java.util.List;
import java.util.Map;

@Endpoint
public class ReviverEndpoint {

  public Date date(Date arg) {
    return arg;
  }

  @Nullable
  public Date nullableDate(@Nullable Date arg) {
    return arg;
  }

  public BigInteger bigInteger(BigInteger arg) {
    return arg;
  }

  public Map<Integer, List<Date>> map(Map<Integer, List<Date>> arg) {
    return arg;
  }

  public Level2 reviveProperties(Level2 arg) {
    return arg;
  }

  public record Level1(String name, Date date) {}

  public record Level2(int id, @Nullable Level1 info) {}
}

File: generator/src/test/resources/com/cromoteca/samples/endpoints/ReviverEndpoint.ts
import client from '../../../../connect-client.default.js';
import type Level2 from './ReviverEndpoint/Level2.js';
import { type EndpointRequestInit } from '@vaadin/hilla-frontend';

async function bigInteger(arg: BigInt, init?: EndpointRequestInit): Promise<BigInt> {
    return client.call('com.cromoteca.samples.endpoints.ReviverEndpoint', 'bigInteger', { arg }, init);
}

async function date(arg: Date, init?: EndpointRequestInit): Promise<Date> {
    return client.call('com.cromoteca.samples.endpoints.ReviverEndpoint', 'date', { arg }, init);
}

async function map(arg: Map<number, Array<Date>>, init?: EndpointRequestInit): Promise<Map<number, Array<Date>>> {
    return client.call('com.cromoteca.samples.endpoints.ReviverEndpoint', 'map', { arg }, init);
}

async function nullableDate(arg: Date | undefined, init?: EndpointRequestInit): Promise<Date | undefined> {
    return client.call('com.cromoteca.samples.endpoints.ReviverEndpoint', 'nullableDate', { arg }, init);
}

async function reviveProperties(arg: Level2, init?: EndpointRequestInit): Promise<Level2> {
    return client.call('com.cromoteca.samples.endpoints.ReviverEndpoint', 'reviveProperties', { arg }, init);
}

const ReviverEndpoint = {
    bigInteger,
    date,
    map,
    nullableDate,
    reviveProperties,
};

export default ReviverEndpoint;
